### ISSUE TITLE
Search results missing instances of strings for certain files unless file is open

### ISSUE THREAD

**Issue by BrandonClapp** [microsoft/vscode#116213]
- VS Code Version: 1.53.0
- OS Version: Windows 10 Version 10.0.18363 Build 18363

Steps to Reproduce:

1. Search for text in a file
2. See the search results. Note that some search results are missing.
3. Open file containing the missing results
4. Search again
5. See the search results. Note that the previously missing results are now found.
6. Close the file containing the newly found results.
7. Search again
8. Note that the results are missing again

Note: This doesn't happen for every file. I'm not sure why this one is special.

https://user-images.githubusercontent.com/2096746/107405595-1bab8e80-6acd-11eb-933a-5eb1ea867e43.mp4

Expected Behavior: The search returns results for all instances of the string, regardless if the file is open or not.

Does this issue occur when all extensions are disabled?: Yes, same result

**Comment by JacksonKearl** [comment]
Does the result appear if you remove the excludes or re-enable excluding standard files? (the minus-gear icon in the excludes filter that is deselected in the video) cc @roblourens

**Comment by BrandonClapp** [comment]
Unfortunately, neither of those seemed to help.

I tried...

- Removed `*node_modules*, *dist*` from the "files to exclude" field.
- Toggled "Use Exclude Settings and Ignore Files" both on/off

I'm also able to reproduce it on a file in a sibling directory. Is it possible that the length of the path to the file could affect search results? The path length character count to this particular file is 267 characters. This is purely a suspicion, though.

**Comment by BrandonClapp** [comment]
The path length turned out to be the issue. I fixed this one off issue by making a symlink closer to my drive and then opening the folder using that symlink path.

If anyone else has this issue, this is how I resolved it:

- Open command prompt (as administrator)
- `mklink /J "C:\Source" "C:\Some\Longer\Base\Path"`
- Reopen VSCode at the shorter path
- Hope your paths are under now under 255 characters

...or, you know, just chill out on your folder structure!
